crates/rerun_c/src/lib.rs

```diff
@@ -584,30 +584,44 @@ fn rr_log_impl(
 
     let mut cells = re_log_types::DataCellVec::default();
     cells.reserve(num_data_cells as usize);
-    for i in 0..num_data_cells {
-        let data_cell: &CDataCell = unsafe { &*data_cells.wrapping_add(i as _) };
+
+    let data_cells = unsafe { std::slice::from_raw_parts(data_cells, num_data_cells as usize) };
+
+    for data_cell in data_cells {
         let CDataCell {
             component_name,
-            num_bytes,
-            bytes,
-        } = *data_cell;
+            array,
+            schema,
+        } = &data_cell;
 
         let component_name = component_name.as_str("data_cells[i].component_name")?;
         let component_name = ComponentName::from(component_name);
 
-        let bytes = unsafe { std::slice::from_raw_parts(bytes, num_bytes as usize) };
-        let array = parse_arrow_ipc_encapsulated_message(bytes).map_err(|err| {
+        let field = unsafe { arrow2::ffi::import_field_from_c(schema) }.map_err(|err| {
             CError::new(
-                CErrorCode::ArrowIpcMessageParsingFailure,
-                &format!("Failed to parse Arrow IPC encapsulated message: {err}"),
+                CErrorCode::ArrowFfiSchemaImportError,
+                &format!("Failed to import ffi schema: {err}"),
             )
         })?;
 
+        // We need to copy the array since `import_array_from_c` takes ownership of the array.
+        // Since we don't touch the original afterwards anymore and array doesn't have a drop implementation,
+        // it should be safe to do so.
+        let ffi_array: arrow2::ffi::ArrowArray = unsafe { std::mem::transmute_copy(array) };
```

Comment:
this isn't quite right. Need to handle this in a safer way. Unclear right now if we're leaking memory or (worse) double free stuff.